R/req-perform.R

```diff
@@ -34,8 +38,22 @@
 #' @examples
 #' request("https://google.com") %>%
 #'   req_perform()
-req_perform <- function(req, path = NULL, verbosity = getOption("httr2_verbosity", 0L)) {
+req_perform <- function(
+      req,
+      path = NULL,
+      verbosity = getOption("httr2_verbosity", 0L),
+      mock = getOption("httr2_mock", NULL)
+  ) {
   check_request(req)
+
+  if (!is.null(mock)) {
+    mock <- as_function(mock)
+    mock_resp <- mock(req)
+    if (!is.null(mock_resp)) {
```

Comment:
Why check for `NULL` specifically? If anything I'd think `stopifnot(is_response(mock_resp))` would make sense, or just let the `mock` function return whatever it wants.

Reply:
So the function can opt out of handling a request

Reply:
Ah I see you said that in the docs. Is there a use case you had in mind for this? It's not something I had considered; I'd think either you're mocking everything or nothing.

Reply:
I dunno, it just seems weird to completely take over all http requests with no way to opt out.

Reply:
Mocking *is* weird ;) I think it would be problematic if you're mocking responses but some requests silently pass through and call out to the internet, so I can't imagine a use case for this. But I guess here you'd have to explicitly design your mock function to return NULL in those cases, so you'd have control over that?

Reply:
Right. I think this design lets you opt out if you want to, but you'd have to deliberately design your function to do so.